hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java

```diff
@@ -875,6 +883,33 @@ public static <T> boolean isDefaultValueDefined(Configuration conf, ConfigOption
         || conf.get(option).equals(option.defaultValue());
   }
 
+  public static String getKeyGenClassNameByType(Configuration conf) {
+    String genType = conf.get(FlinkOptions.KEYGEN_TYPE);
```

Comment:
Can giving the option `KEYGEN_CLASS_NAME` a default value: `SimpleAvroKeyGenerator`
solves your problem ?

Reply:
If you set the `KEYGEN_CLASS_NAME` default value, it does solve the problem of not showing.

However, in the original logic, i`` KEYGEN_CLASS_NAME` is not set, `KEYGEN_TYPE` will be read.
Therefore, if the default value is set for `KEYGEN_CLASS_NAME`, the `KEYGEN_TYPE` configuration is equivalent to completely useless.

Reply:
Then why we still always set up the param `KEYGEN_CLASS_NAME` in this patch for SQL then ?

Reply:
If `KEYGEN_CLASS_NAME` is not set, this field will not be displayed in the table properties. 
`KEYGEN_TYPE` is equivalent to an alias of `KEYGEN_CLASS_NAME`, so `KEYGEN_TYPE` can be read here to initialize `KEYGEN_CLASS_NAME`. And the default value of `KEYGEN_TYPE` is `SIMPLE`, so `KEYGEN_CLASS_NAME` will be set to `SimpleAvroKeyGenerator` by default.

According to the above logic, if the user does not set `KEYGEN_CLASS_NAME`, `KEYGEN_CLASS_NAME` can be replaced by setting `KEYGEN_TYPE`, and after setting, `KEYGEN_CLASS_NAME` will also be stored in the table propeties. In this way, the function of `KEYGEN_TYPE` not only conforms to the original logic, but also conforms to the description in the document.

Reply:
So what are we fixing here if `KEYGEN_TYPE` default value is set up and a keygen clazz can be inferred correctly.

Reply:
If there is no current process, then `KEYGEN_CLASS_NAME` will not be assigned, and there will be no KEYGEN_CLASS_NAME property in table properties.
If you set the default value of `KEYGEN_CLASS_NAME` as you said above, then the `KEYGEN_TYPE` parameter will be invalid.
Therefore, I initialized `KEYGEN_CLASS_NAME` according to `KEYGEN_TYPE`, so that not only can the value of `KEYGEN_CLASS_NAME` be saved in table properties, but `KEYGEN_TYPE` can also take effect normally according to the original logic.